Frameworks/CoreGraphics/CGColor.mm

```diff
@@ -91,97 +216,72 @@ CGColorRef CGColorCreateCopy(CGColorRef color) {
 /**
  @Status Interoperable
 */
-CGColorRef CGColorCreateCopyWithAlpha(CGColorRef color, float alpha) {
+CGColorRef CGColorCreateCopyWithAlpha(CGColorRef color, CGFloat alpha) {
     RETURN_NULL_IF(!color);
-    static __CGColorQuad defaultColor{ 0.0f, 0.0f, 0.0f, 0.0f };
-    const __CGColorQuad* curColor = [(UIColor*)color _getColors];
-    if (!curColor) {
-        curColor = &defaultColor;
-    }
-
-    CGColorRef ret =
-        static_cast<CGColorRef>([[__LazyUIColor colorWithRed:curColor->r green:curColor->g blue:curColor->b alpha:alpha] retain]);
-    [static_cast<UIColor*>(ret) setColorSpace:CGColorGetColorSpace(color)];
-    return ret;
-}
-
-/**
- @Status Caveat
- @Notes components are not supported
-*/
-CGColorRef CGColorCreateWithPattern(CGColorSpaceRef colorSpace, CGPatternRef pattern, const CGFloat* components) {
-    RETURN_NULL_IF(!colorSpace);
-    RETURN_NULL_IF(!pattern);
-    CGColorRef ret = static_cast<CGColorRef>([[__LazyUIColor _colorWithCGPattern:(CGPatternRef)pattern] retain]);
-    [static_cast<UIColor*>(ret) setColorSpace:colorSpace];
-    return ret;
+    return __CGColor::CreateInstance(kCFAllocatorDefault, color, alpha);
 }
 
 /**
  @Status Interoperable
 */
 bool CGColorEqualToColor(CGColorRef color1, CGColorRef color2) {
-    if (color1 == color2) {
-        return true;
-    }
-    if (!color1 || !color2) {
-        return false;
-    }
-    const __CGColorQuad* components1 = [(UIColor*)color1 _getColors];
-    const __CGColorQuad* components2 = [(UIColor*)color2 _getColors];
-
-    return ((components1->r == components2->r) && (components1->g == components2->g) && (components1->b == components2->b) &&
-            (components1->a == components2->a)) &&
-           (CGColorSpaceGetModel(CGColorGetColorSpace(color1)) == CGColorSpaceGetModel(CGColorGetColorSpace(color2)));
+    RETURN_RESULT_IF(color1 == color2, true);
```

Comment:
early return would void the call into CFEqual

---
In reply to: [111798718](https://github.com/Microsoft/WinObjC/pull/2538#discussion_r111798718) [](ancestors = 111798718)